### ISSUE TITLE
Mutation during displacement

### ISSUE THREAD

**Issue by xiangWu-WW** [yjhp1016/taichi_LBM3D#30]
![iter-5000](https://github.com/user-attachments/assets/b5f55df4-53bd-4969-b585-c78450f03526)
![iter-20000](https://github.com/user-attachments/assets/a4597003-3f2b-4cc8-a9fd-edd560c873b9)
![iter-40000](https://github.com/user-attachments/assets/d184d34b-9edf-4bdf-ae13-5795754cb86a)
![iter-75000](https://github.com/user-attachments/assets/cdeda6dc-8a0c-48f6-8ff4-f7760fab0f5b)
![iter-80000](https://github.com/user-attachments/assets/2be27007-a200-47c1-a4fe-ff0a8f4efa77)
<img width="728" alt="e4d6c01837531afa001fa07afa48ee1" src="https://github.com/user-attachments/assets/3926c86c-1d51-47cc-8455-e624cf8704be">
Hello, Dr. Yang. I am new to CFD, I am doing two-phase flow displacement, I built a 64^3 model, apply the driving force in the left side of the X direction, the displacement was slow for the first 75,000 times, by 80,000 times it was all through, They are 5,000, 20,000, 40,000, 75,000 and 80,000 times respectively，what is the cause of this?
In addition, the phase_ftb64.dat file is all set to 1, and the fixed phase boundary condition is used. Then, does the parameter setting of niu_g represent -1 is the phase of displacement force, and does psi_solid represent the contact Angle of niu_g phase between niu_l and the solid skeleton surface? Looking forward to your reply.

**Comment by yjhp1016** [comment]
Yes, psi_solid is the cosine of desired contact angle between non wetting
phase and surface.
It looks like the viscosity of niu_g is quite big, probably that's why the
invasion was slow
if you can make Y and Z direction psi boundary conditions to 0 (periodic)
then re-run, you probably will get something different.

This is a small geometry, if we constrain X inlet with non-wetting and
constrain all rest 5 faces as wetting phase, this will make the invasion
quite difficult, so we can "open" Y and Z direction to let the fluid has
more freedom to move, I hope this change might improve the simulation.
